### ISSUE TITLE
about batch_size and seq_len

### ISSUE THREAD

**Issue by xiaocainiao** [jaychanhoi/Lattice_LSTM#2]
Hi~
I have some questions about trainning process.
Does the model only process fixed batch_size and sequence_length? How to process dynamic batches?

When running the last batch (less than batch_size),  I get an error: 

> ValueError: The shape for char-lexicon/rnn/while/Merge_2:0 is not an invariant for the loop. It enters the loop with shape (64, 50), but has shape <unknown> after one iteration. Provide shape invariants using either the `shape_invariants` argument of tf.while_loop or set_shape() on the loop variables.

If I get the dynamic batch_size of each batch and feed into LatticeCell, I get an error like 

> ValueError: Cannot convert a partially known TensorShape to a Tensor: (?, 50)

**Comment by JayChanHoi** [comment]
Hi~

Yea. Now, It only support fixed batch size and seq_len. The reason for this is that at the time I develop this, Tensorflow don’t allow me to make the batch size dynamic for tf.scatter_nd_update() as it can only take variable. And Tensorflow don’t support for dynamic shape of variables as the methodology of Tensorflow is to build an static graph first followed with run. If you wish to use it as dynamic batch size and seq_len, a good choice is to make this model by using PyTorch, which is dynamic.

**Comment by weilongheal** [comment]
I don't think so, TensorFlow supports dynamic RNN(LSTM) sequence length, like: tf.nn.bidirectional_dynamic_rnn. 

Thank you a lot first ! Before that, I also want to convert the paper to tensorflow-based.
@chan111222333 

Could'you give us the version you used:python, tensorflow?

**Comment by JayChanHoi** [comment]
Yea. tf does support dynamic arguments in some situation, but it doesn’t for the scatter_update. There maybe some other kinds way to do as scatter_update. I think for the recent version of tf or the coming 2.0, it may be able to have a dynamic version for lattice-lstm. You definitely should try to do it if you want to.:) 

Python should be 3.6 and tensorflow should be 1.8.

從我的 iPhone 傳送

**Comment by qibaoyuan** [comment]
> I don't think so, TensorFlow supports dynamic RNN(LSTM) sequence length, like: tf.nn.bidirectional_dynamic_rnn.
> 
> Thank you a lot first ! Before that, I also want to convert the paper to tensorflow-based.
> @chan111222333
> 
> Could'you give us the version you used:python, tensorflow?

hello, I am wondering if you have solved the problem and convert the source code from pytorch to tensorflow version?
